plugins/node/opentelemetry-instrumentation-redis/src/types.ts

```diff
@@ -36,4 +37,43 @@ export interface RedisPluginClientTypes {
   address?: string;
 }
 
-export type RedisInstrumentationConfig = InstrumentationConfig;
+/**
+ * Function that can be used to serialize db.statement tag
+ * @param cmdName - The name of the command (eg. set, get, mset)
+ * @param cmdArgs - Array of arguments passed to the command
+ *
+ * @returns serialized string that will be used as the db.statement attribute.
+ */
+export type DbStatementSerializer = (
+  cmdName: RedisCommand['command'],
+  cmdArgs: RedisCommand['args']
+) => string;
+
+/**
+ * Function that can be used to add custom attributes to span on response from redis server
+ * @param span - The span created for the redis command, on which attributes can be set
+ * @param cmdName - The name of the command (eg. set, get, mset)
+ * @param cmdArgs - Array of arguments passed to the command
+ * @param response - The response object which is returned to the user who called this command.
+ *  Can be used to set custom attributes on the span.
+ *  The type of the response varies depending on the specific command.
+ */
+export interface RedisResponseCustomAttributeFunction {
```

Comment:
Is there a reason why this is in `interface` but `DbStatementSerializer` a `type`?

Reply:
No particular reason, just keeping the same approach as in the io-redis instrumentation